services/nexus/nexus.service.js

```diff
@@ -46,7 +46,7 @@ module.exports = class Nexus extends BaseJsonService {
       pattern:
         // Do not base new services on this route pattern.
         // See https://github.com/badges/shields/issues/3714
-        ':repo(r|s|[^/]+)/:scheme(http|https)/:hostAndPath+/:groupId/:artifactId([^/:]+):queryOpt(:.+)?',
+        ':repo(r|s|[^/]+)/:scheme(http|https)/:hostAndPath+/:groupId/:artifactId([^/:]+?):queryOpt(:.+?)?',
```

Comment:
I'm getting a `404 badge not found` on these nexus badges when trying to use the optional `queryOpt`, although that's the same behavior in prod 🤔 

It's service patterns like this that I think will really benefit from moving the instance/server host url to a query param, especially when the instance/server can have an optional path segment and the badge route has a trailing optional param

Reply:
Hmm, probably we should add a test. I leaned pretty heavily on the tests for these. Can you give me an example of a URL that should work?

Reply:
I was just using the values we have listed in the example for the `Sonatype Nexus (Query Options)` listing. Altogether, that generates the below badge url (from prod):

https://img.shields.io/nexus/fs-public-snapshots/https/repository.jboss.org/nexus/com.progress.fuse/fusehq/:c=agent-apple-osx:p=tar.gz.svg
![](https://img.shields.io/nexus/fs-public-snapshots/https/repository.jboss.org/nexus/com.progress.fuse/fusehq/:c=agent-apple-osx:p=tar.gz.svg)

Reply:
That URL is 404 in prod too, though if you remove the `/` before `:c=agent-apple-osx` it works:

https://img.shields.io/nexus/fs-public-snapshots/https/repository.jboss.org/nexus/com.progress.fuse/fusehq:c=agent-apple-osx:p=tar.gz.svg

Reply:
> That URL is 404 in prod too,

Yup, not related to this PR but worth digging into. Maybe that example listing could be updated in the nexus service to combine the artifactId and query into a single route param?

I'll open a new issue to discuss/track

Reply:
Ahh gotcha. Maybe the optional queryOpt should be moved into a query param, too.

Reply:
That sounds like a good fix!